_src/_assets/js/dashboard-map.js

```diff
@@ -134,68 +202,88 @@
     // })
 
     function updateLegend() {
-      if (useChloropleth) {
+      //miggt be better to add both legends once and toggle
+      d3.select('#map-legend')
+        .selectAll('*')
+        .remove()
+      if (usechoropleth) {
         d3.select('#map-legend')
-          .selectAll('*')
-          .remove()
-        return
+          .append('span')
+          .attr('style', 'font-weight: 600')
+          .text(`per million residents`)
+        d3.select('#map-legend').append(() =>
+          d3Legend({
+            color: getColor[currentField],
+            height: 40,
+            width: 300,
+            marginTop: 8,
+            tickFormat: '~s',
+            spaceBetween: 2,
+            tickSize: 0,
+          }),
+        )
+      } else {
+        //bubble legend
+        const legend = d3
```

Comment:
Looking forward to getting this refactored in the `d3Legend` so we can use it in two places 🎉 🎉 🎉

Reply:
agreed